scipy/stats/tests/test_kdeoth.py

```diff
@@ -318,6 +318,51 @@ def test_kde_integer_input():
     assert_array_almost_equal(kde(x1), y_expected, decimal=6)
 
 
+@pytest.mark.parametrize("bw_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']] +
+                         ["scott", "silverman"])
+@pytest.mark.parametrize("weights_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']])
+@pytest.mark.parametrize("dataset_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']])
+@pytest.mark.parametrize("point_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']])
+def test_kde_output_dtype(point_type, dataset_type, weights_type, bw_type):
+    # test that for any given combination of input datatypes we get the 
+    # appropriate result datatype
+    if point_type is None:
+        pytest.skip("Data type not available")
+    if dataset_type is None:
+        pytest.skip("Data type not available")
+
+    if weights_type is None:
+        pytest.skip("Data type not available")
+    else:
+        weights = np.arange(5, dtype=weights_type)
+
+    if bw_type is None:
+        pytest.skip("Data type not available")
+    elif bw_type in ["scott", "silverman"]:
+        bw = bw_type
+    else:
+        bw = bw_type(3)
+
+    dataset = np.arange(5, dtype=dataset_type)
+    k = stats.kde.gaussian_kde(dataset, bw_method=bw, weights=weights)
+    points = np.arange(5, dtype=point_type)
+    result = k(points)
+    assert result.dtype == np.result_type(dataset, points, weights, k.factor)
+
+def test_kde_output_dtype():
+    # test that kde result is floating if input is not floating
+    test_type = np.int
```

Comment:
As before, don't using `np.int`.  In fact, it looks like you can remove this line, and just use `points = np.arange(5)` below.

Reply:
In fact, I think that test can just be combined with the other one, now that we're using the common type. That should cover integer datatypes in the same way as floats!